packages/backend/src/accounting/service.ts

```diff
@@ -133,30 +131,32 @@ export async function createLiquidityAccount(
     {
       id: account.id,
       type: AccountType.Credit,
-      unit: account.asset.unit
+      ledger: account.asset.ledger,
+      code: AccountCode.Liquidity
     }
   ])
   return account
 }
 
 export async function createSettlementAccount(
   deps: ServiceDependencies,
-  unit: number
+  ledger: number
 ): Promise<void> {
   try {
     await createAccounts(deps, [
       {
-        id: unit,
+        id: ledger,
         type: AccountType.Debit,
-        unit
+        ledger,
+        code: AccountCode.Settlement
       }
     ])
   } catch (err) {
     // Don't complain if asset settlement account already exists.
     // This could change if TigerBeetle could be reset between tests.
     if (
       err instanceof CreateAccountError &&
-      err.code === CreateAccountErrorCode.exists
+      areAllAccountExistsErrors(err.codes)
```

Comment:
If the error is one of:

- `exists_with_different_flags`
- `exists_with_different_user_data`
- `exists_with_different_ledger`
- `exists_with_different_code`

I think that it is an actual error that needs to be surfaced to the Rafiki operation. Those fields shouldn't ever change between `create_account` attempts.